routers/repo/editor.go

```diff
@@ -676,6 +677,156 @@ func UploadFilePost(ctx *context.Context, form auth.UploadRepoFileForm) {
 	}
 }
 
+// UploadFilePostJson JSON response for uploading file
+func UploadFilePostJson(ctx *context.Context, form auth.UploadRepoFileForm) {
+	// `canCommit` is a poorly named; it's not responsible directly for rendering anything,
+	// it returns a boolean with commit rights
+	canCommit := renderCommitRights(ctx)
```

Comment:
Seems like a perfect name to me? What do you mean by "responsible for rendering anything", what could it be rendering in this context?